controllers/subscription_controller.go

```diff
@@ -310,6 +312,18 @@ func (r *SubscriptionReconciler) setOperatorCondition(ctx context.Context, logge
 		}
 	}
 
+	// Make operator not upgradeable if ODF version is n+1 ahead of OCP version
+	isODFAhead, err := isODFAheadOfOCP(ctx, r.Client, r.operatorConditionName)
+	if err != nil {
+		return err
+	}
+	if isODFAhead {
+		msg := "ODF version is already ahead of OCP. Further ODF upgrade would make it incompatible with the current OCP version"
```

Comment:
And why aren't we enforcing ODF version resulting in ODF being already ahead of OCP?